### DISCUSSION TITLE
Does IMage::write_to_tofile implicitly convert to sRGB?

### DISCUSSION THREAD

**Discussion by VivitionDeveloper** [libvips/libvips#4130] [General]
I discovered that writing a vips image in LABS colorspace to disk produces same output as when first a convert to sRGB is done. Does this mean that VIMage::write_to_tofile() implicitly converts an image to sRGB?

I've checked [vips_foreign_find_save](https://www.[libvips](https://www.libvips.org/API/current/VipsForeignSave.html#vips-foreign-find-save).org/API/current/VipsForeignSave.html#vips-foreign-find-save) and [vips_image_write_to_file](https://www.[libvips](https://www.libvips.org/API/current/VipsImage.html#vips-image-write-to-file).org/API/current/VipsImage.html#vips-image-write-to-file), but did not find information about that.

I'm not very familiar to the JPEG format, but according [Wiki ](https://en.wikipedia.org/wiki/JPEG_File_Interchange_Format#Color_space) the color space can be written to the file (remembering things about sRGB vs Adobe RGB when printing photos).

Actually, the code behind this question is how to adjust saturation. Borrowed from another [example ](https://github.com/libvips/libvips/issues/1662) I created some code like this (thanks to another piece of [information](https://github.com/libvips/libvips/discussions/4036), that direct convert to LABS is not possible):

```
img = img.colourspace(VipsInterpretation::VIPS_INTERPRETATION_LABQ)
		.colourspace(viInterpretation);
// adjust saturation by multiplying the AB channels and leave L untouched (=0)
img = img.linear(1, { 0 , 1 + dAmount, 1 + dAmount });

img.write_to_file(sStoreFileName);
```

Actually two questions:

1. does the write implicitly perform a convert to sRGB
2. I did not notice much/any difference when chaning the AB channels, e.g. with 1.3. What am I doing wrong?

**Comment by jcupitt** [comment]
Hello @CoachRDeveloper,

Yes, the savers will implicitly convert the image to one of the formats supported by the image type. So (for example) the jpeg saver will convert to one of: one band mono, three band RGB, or four band CMYK, all of then 8 bit depth.

For your code, try (python for brevity):

```python
img = img.colourspace("lab")
img *= [1, 1.3, 1.3]
img.write_to_file("xx.jpg")
```

`labq` is a strange 32 bit encoded format and you can't do arithmetic on it directly. You want plain `lab`, which is 3 float bands and very easy to manipulate. The jpeg saver will automatically convert to RGB for you, or you can do the conversion yourself, of course.

**Reply by VivitionDeveloper** [reply]
Thanks about confirming the behavior of the write. May be something that is worth to be added to the documentation?

I know about  the LABQ issue. That was mentioned by you in another discussion (and according to previous comments, to be fixed in 8.16). I did not use LABQ, but LABS (VipsInterpretation::VIPS_INTERPRETATION_LABS).

I wonder whether I did something wrong in my C++ code about lineair. According your python example (I'm not familiar with it) it looks about the same. According to the documentation, img.linear multiplies double a to the image and add b. That's probably my mistake. I have changed my code to:

```
img = img.linear({ 1 , 1 + dAmount, 1 + dAmount }, 0);
````
